Synthesis failed for python-access-context-manager

Hello! Autosynth couldn't regenerate python-access-context-manager. :broken_heart:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
> Running: git log -1 --pretty=%at 3713f07b86adf7578ef24f0c456f5ff38dae85d2
2021-04-03 03:22:12,951 autosynth [DEBUG] > Running: git log -1 --pretty=%at 83b8f9c243f22a20b86c9b3e19e8c214b69aea97
2021-04-03 03:22:12,954 autosynth [DEBUG] > Running: git log -1 --pretty=%at 328870c3f64c64d4576ac1d6e406e88acce4e58a
2021-04-03 03:22:12,957 autosynth [DEBUG] > Running: git log -1 --pretty=%at 572ef8f70edd9041f5bcfa71511aed6aecfc2098
2021-04-03 03:22:12,960 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4dc31ac1ece23dc555f574aa701e5857b5bf2d3f
2021-04-03 03:22:12,963 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4501974ad08b5d693311457e2ea4ce845676e329
2021-04-03 03:22:12,966 autosynth [DEBUG] > Running: git log -1 --pretty=%at e8d9c5ebf915069df34c207ba8b9a1cfbdb39592
2021-04-03 03:22:12,969 autosynth [DEBUG] > Running: git log -1 --pretty=%at 551dd78ca04f7989abc9e63e392f8b8cfa1a0ef9
2021-04-03 03:22:12,972 autosynth [DEBUG] > Running: git log -1 --pretty=%at 68b8d39a3d5338ed5ff068ad8aa59fb271ef088b
2021-04-03 03:22:12,975 autosynth [DEBUG] > Running: git log -1 --pretty=%at edbd72054bf9f71505782a94d9973e26866f7038
2021-04-03 03:22:12,978 autosynth [DEBUG] > Running: git log -1 --pretty=%at f6132d5e7eddc9d8f08bc4a1cca9234baee38c94
2021-04-03 03:22:12,980 autosynth [DEBUG] > Running: git log -1 --pretty=%at a22531f8364582f51485e4e9db85872d93514ab1
2021-04-03 03:22:12,983 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-03 03:22:12,986 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-03 03:22:12,990 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-03 03:22:12,995 autosynth [DEBUG] > Running: git checkout ff39353f34a36e7643b86e97724e4027ab466dc6
Note: checking out 'ff39353f34a36e7643b86e97724e4027ab466dc6'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at ff39353 fix: non-root users can run the nodejs docker post processor (#1016)
2021-04-03 03:22:13,003 autosynth [DEBUG] > Running: git branch -f autosynth-21
2021-04-03 03:22:13,006 autosynth [DEBUG] > Running: git checkout autosynth-21
Switched to branch 'autosynth-21'
2021-04-03 03:22:13,012 autosynth [INFO] > Running synthtool
2021-04-03 03:22:13,012 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-03 03:22:13,012 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/21/sponge_log.log
2021-04-03 03:22:13,014 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-03 03:22:13,243 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-21
nothing to commit, working tree clean
2021-04-03 03:22:13,379 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-03 03:22:13,385 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-03 03:22:13,770 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-03 03:22:13,841 autosynth [ERROR] > Synthesis failed
2021-04-03 03:22:13,842 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-03 03:22:13,849 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-03 03:22:13,855 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/48bd838b-17eb-4c50-ba23-29b42af5d6fc/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
> Running: git log -1 --pretty=%at 3713f07b86adf7578ef24f0c456f5ff38dae85d2
2021-04-04 03:21:22,995 autosynth [DEBUG] > Running: git log -1 --pretty=%at 83b8f9c243f22a20b86c9b3e19e8c214b69aea97
2021-04-04 03:21:22,997 autosynth [DEBUG] > Running: git log -1 --pretty=%at 328870c3f64c64d4576ac1d6e406e88acce4e58a
2021-04-04 03:21:23,000 autosynth [DEBUG] > Running: git log -1 --pretty=%at 572ef8f70edd9041f5bcfa71511aed6aecfc2098
2021-04-04 03:21:23,003 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4dc31ac1ece23dc555f574aa701e5857b5bf2d3f
2021-04-04 03:21:23,006 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4501974ad08b5d693311457e2ea4ce845676e329
2021-04-04 03:21:23,008 autosynth [DEBUG] > Running: git log -1 --pretty=%at e8d9c5ebf915069df34c207ba8b9a1cfbdb39592
2021-04-04 03:21:23,011 autosynth [DEBUG] > Running: git log -1 --pretty=%at 551dd78ca04f7989abc9e63e392f8b8cfa1a0ef9
2021-04-04 03:21:23,014 autosynth [DEBUG] > Running: git log -1 --pretty=%at 68b8d39a3d5338ed5ff068ad8aa59fb271ef088b
2021-04-04 03:21:23,017 autosynth [DEBUG] > Running: git log -1 --pretty=%at edbd72054bf9f71505782a94d9973e26866f7038
2021-04-04 03:21:23,019 autosynth [DEBUG] > Running: git log -1 --pretty=%at f6132d5e7eddc9d8f08bc4a1cca9234baee38c94
2021-04-04 03:21:23,022 autosynth [DEBUG] > Running: git log -1 --pretty=%at a22531f8364582f51485e4e9db85872d93514ab1
2021-04-04 03:21:23,025 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-04 03:21:23,028 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-04 03:21:23,031 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-04 03:21:23,036 autosynth [DEBUG] > Running: git checkout ff39353f34a36e7643b86e97724e4027ab466dc6
Note: checking out 'ff39353f34a36e7643b86e97724e4027ab466dc6'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at ff39353 fix: non-root users can run the nodejs docker post processor (#1016)
2021-04-04 03:21:23,043 autosynth [DEBUG] > Running: git branch -f autosynth-21
2021-04-04 03:21:23,045 autosynth [DEBUG] > Running: git checkout autosynth-21
Switched to branch 'autosynth-21'
2021-04-04 03:21:23,049 autosynth [INFO] > Running synthtool
2021-04-04 03:21:23,049 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-04 03:21:23,049 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/21/sponge_log.log
2021-04-04 03:21:23,051 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-04 03:21:23,270 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-21
nothing to commit, working tree clean
2021-04-04 03:21:23,401 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-04 03:21:23,406 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-04 03:21:23,763 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-04 03:21:23,822 autosynth [ERROR] > Synthesis failed
2021-04-04 03:21:23,822 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-04 03:21:23,830 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-04 03:21:23,835 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/1c3c3150-3af8-4023-9da4-b625d4ddec58/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
> Running: git log -1 --pretty=%at 3713f07b86adf7578ef24f0c456f5ff38dae85d2
2021-04-05 03:23:14,619 autosynth [DEBUG] > Running: git log -1 --pretty=%at 83b8f9c243f22a20b86c9b3e19e8c214b69aea97
2021-04-05 03:23:14,622 autosynth [DEBUG] > Running: git log -1 --pretty=%at 328870c3f64c64d4576ac1d6e406e88acce4e58a
2021-04-05 03:23:14,625 autosynth [DEBUG] > Running: git log -1 --pretty=%at 572ef8f70edd9041f5bcfa71511aed6aecfc2098
2021-04-05 03:23:14,628 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4dc31ac1ece23dc555f574aa701e5857b5bf2d3f
2021-04-05 03:23:14,631 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4501974ad08b5d693311457e2ea4ce845676e329
2021-04-05 03:23:14,634 autosynth [DEBUG] > Running: git log -1 --pretty=%at e8d9c5ebf915069df34c207ba8b9a1cfbdb39592
2021-04-05 03:23:14,637 autosynth [DEBUG] > Running: git log -1 --pretty=%at 551dd78ca04f7989abc9e63e392f8b8cfa1a0ef9
2021-04-05 03:23:14,641 autosynth [DEBUG] > Running: git log -1 --pretty=%at 68b8d39a3d5338ed5ff068ad8aa59fb271ef088b
2021-04-05 03:23:14,644 autosynth [DEBUG] > Running: git log -1 --pretty=%at edbd72054bf9f71505782a94d9973e26866f7038
2021-04-05 03:23:14,647 autosynth [DEBUG] > Running: git log -1 --pretty=%at f6132d5e7eddc9d8f08bc4a1cca9234baee38c94
2021-04-05 03:23:14,650 autosynth [DEBUG] > Running: git log -1 --pretty=%at a22531f8364582f51485e4e9db85872d93514ab1
2021-04-05 03:23:14,653 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-05 03:23:14,657 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-05 03:23:14,660 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-05 03:23:14,666 autosynth [DEBUG] > Running: git checkout ff39353f34a36e7643b86e97724e4027ab466dc6
Note: checking out 'ff39353f34a36e7643b86e97724e4027ab466dc6'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at ff39353 fix: non-root users can run the nodejs docker post processor (#1016)
2021-04-05 03:23:14,673 autosynth [DEBUG] > Running: git branch -f autosynth-21
2021-04-05 03:23:14,676 autosynth [DEBUG] > Running: git checkout autosynth-21
Switched to branch 'autosynth-21'
2021-04-05 03:23:14,682 autosynth [INFO] > Running synthtool
2021-04-05 03:23:14,682 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-05 03:23:14,682 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/21/sponge_log.log
2021-04-05 03:23:14,684 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-05 03:23:14,918 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-21
nothing to commit, working tree clean
2021-04-05 03:23:15,061 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-05 03:23:15,067 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-05 03:23:15,449 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-05 03:23:15,503 autosynth [ERROR] > Synthesis failed
2021-04-05 03:23:15,503 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-05 03:23:15,513 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-05 03:23:15,518 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/1c3f7a04-7c3d-443e-867e-346507a267ee/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
G] > Running: git log -1 --pretty=%at 328870c3f64c64d4576ac1d6e406e88acce4e58a
2021-04-06 03:23:19,231 autosynth [DEBUG] > Running: git log -1 --pretty=%at 572ef8f70edd9041f5bcfa71511aed6aecfc2098
2021-04-06 03:23:19,233 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4dc31ac1ece23dc555f574aa701e5857b5bf2d3f
2021-04-06 03:23:19,236 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4501974ad08b5d693311457e2ea4ce845676e329
2021-04-06 03:23:19,239 autosynth [DEBUG] > Running: git log -1 --pretty=%at e8d9c5ebf915069df34c207ba8b9a1cfbdb39592
2021-04-06 03:23:19,242 autosynth [DEBUG] > Running: git log -1 --pretty=%at 551dd78ca04f7989abc9e63e392f8b8cfa1a0ef9
2021-04-06 03:23:19,244 autosynth [DEBUG] > Running: git log -1 --pretty=%at 68b8d39a3d5338ed5ff068ad8aa59fb271ef088b
2021-04-06 03:23:19,247 autosynth [DEBUG] > Running: git log -1 --pretty=%at edbd72054bf9f71505782a94d9973e26866f7038
2021-04-06 03:23:19,250 autosynth [DEBUG] > Running: git log -1 --pretty=%at f6132d5e7eddc9d8f08bc4a1cca9234baee38c94
2021-04-06 03:23:19,253 autosynth [DEBUG] > Running: git log -1 --pretty=%at a22531f8364582f51485e4e9db85872d93514ab1
2021-04-06 03:23:19,255 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-06 03:23:19,258 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-06 03:23:19,261 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-06 03:23:19,264 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-06 03:23:19,267 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-06 03:23:19,272 autosynth [DEBUG] > Running: git checkout 063de45298fbdd88916018ba566c7ecd254b39ae
Note: checking out '063de45298fbdd88916018ba566c7ecd254b39ae'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 063de45 chore: Use python 3.6.13-buster for owlbot post-processor (#1022)
2021-04-06 03:23:19,279 autosynth [DEBUG] > Running: git branch -f autosynth-23
2021-04-06 03:23:19,282 autosynth [DEBUG] > Running: git checkout autosynth-23
Switched to branch 'autosynth-23'
2021-04-06 03:23:19,286 autosynth [INFO] > Running synthtool
2021-04-06 03:23:19,286 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-06 03:23:19,286 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/23/sponge_log.log
2021-04-06 03:23:19,288 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-06 03:23:19,513 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-23
nothing to commit, working tree clean
2021-04-06 03:23:19,646 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-06 03:23:19,651 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-06 03:23:20,013 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-06 03:23:20,084 autosynth [ERROR] > Synthesis failed
2021-04-06 03:23:20,084 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-06 03:23:20,092 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-06 03:23:20,097 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/0f420001-537a-443f-b704-5a3b73254f36/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
autosynth [DEBUG] > Running: git log -1 --pretty=%at 4501974ad08b5d693311457e2ea4ce845676e329
2021-04-07 03:25:19,512 autosynth [DEBUG] > Running: git log -1 --pretty=%at e8d9c5ebf915069df34c207ba8b9a1cfbdb39592
2021-04-07 03:25:19,515 autosynth [DEBUG] > Running: git log -1 --pretty=%at 551dd78ca04f7989abc9e63e392f8b8cfa1a0ef9
2021-04-07 03:25:19,518 autosynth [DEBUG] > Running: git log -1 --pretty=%at 68b8d39a3d5338ed5ff068ad8aa59fb271ef088b
2021-04-07 03:25:19,521 autosynth [DEBUG] > Running: git log -1 --pretty=%at edbd72054bf9f71505782a94d9973e26866f7038
2021-04-07 03:25:19,524 autosynth [DEBUG] > Running: git log -1 --pretty=%at f6132d5e7eddc9d8f08bc4a1cca9234baee38c94
2021-04-07 03:25:19,527 autosynth [DEBUG] > Running: git log -1 --pretty=%at a22531f8364582f51485e4e9db85872d93514ab1
2021-04-07 03:25:19,530 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-07 03:25:19,533 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-07 03:25:19,536 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-07 03:25:19,539 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-07 03:25:19,542 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-07 03:25:19,545 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-07 03:25:19,548 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-07 03:25:19,551 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-07 03:25:19,557 autosynth [DEBUG] > Running: git checkout 705743e66f5c0b24a95f7f30619c9d3ef747b317
Note: checking out '705743e66f5c0b24a95f7f30619c9d3ef747b317'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 705743e feat: expose get_staging_dirs() for Python (#1024)
2021-04-07 03:25:19,565 autosynth [DEBUG] > Running: git branch -f autosynth-26
2021-04-07 03:25:19,568 autosynth [DEBUG] > Running: git checkout autosynth-26
Switched to branch 'autosynth-26'
2021-04-07 03:25:19,574 autosynth [INFO] > Running synthtool
2021-04-07 03:25:19,574 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-07 03:25:19,574 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/26/sponge_log.log
2021-04-07 03:25:19,576 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-07 03:25:19,813 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-26
nothing to commit, working tree clean
2021-04-07 03:25:19,955 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-07 03:25:19,961 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-07 03:25:20,359 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-07 03:25:20,425 autosynth [ERROR] > Synthesis failed
2021-04-07 03:25:20,425 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-07 03:25:20,435 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-07 03:25:20,441 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/4abe0344-686d-46e5-982b-a65bff3234ca/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
autosynth [DEBUG] > Running: git log -1 --pretty=%at 4501974ad08b5d693311457e2ea4ce845676e329
2021-04-08 05:38:02,315 autosynth [DEBUG] > Running: git log -1 --pretty=%at e8d9c5ebf915069df34c207ba8b9a1cfbdb39592
2021-04-08 05:38:02,319 autosynth [DEBUG] > Running: git log -1 --pretty=%at 551dd78ca04f7989abc9e63e392f8b8cfa1a0ef9
2021-04-08 05:38:02,322 autosynth [DEBUG] > Running: git log -1 --pretty=%at 68b8d39a3d5338ed5ff068ad8aa59fb271ef088b
2021-04-08 05:38:02,326 autosynth [DEBUG] > Running: git log -1 --pretty=%at edbd72054bf9f71505782a94d9973e26866f7038
2021-04-08 05:38:02,330 autosynth [DEBUG] > Running: git log -1 --pretty=%at f6132d5e7eddc9d8f08bc4a1cca9234baee38c94
2021-04-08 05:38:02,334 autosynth [DEBUG] > Running: git log -1 --pretty=%at a22531f8364582f51485e4e9db85872d93514ab1
2021-04-08 05:38:02,337 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-08 05:38:02,341 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-08 05:38:02,345 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-08 05:38:02,349 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-08 05:38:02,353 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-08 05:38:02,356 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-08 05:38:02,361 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-08 05:38:02,365 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-08 05:38:02,371 autosynth [DEBUG] > Running: git checkout 705743e66f5c0b24a95f7f30619c9d3ef747b317
Note: checking out '705743e66f5c0b24a95f7f30619c9d3ef747b317'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 705743e feat: expose get_staging_dirs() for Python (#1024)
2021-04-08 05:38:02,465 autosynth [DEBUG] > Running: git branch -f autosynth-26
2021-04-08 05:38:02,469 autosynth [DEBUG] > Running: git checkout autosynth-26
Switched to branch 'autosynth-26'
2021-04-08 05:38:02,474 autosynth [INFO] > Running synthtool
2021-04-08 05:38:02,475 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-08 05:38:02,475 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/26/sponge_log.log
2021-04-08 05:38:02,479 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-08 05:38:02,721 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-26
nothing to commit, working tree clean
2021-04-08 05:38:02,885 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-08 05:38:02,891 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-08 05:38:03,310 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-08 05:38:03,381 autosynth [ERROR] > Synthesis failed
2021-04-08 05:38:03,381 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-08 05:38:03,390 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-08 05:38:03,397 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/99375984-21c6-4fad-af8b-1c6643e5f29a/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
Running: git log -1 --pretty=%at 551dd78ca04f7989abc9e63e392f8b8cfa1a0ef9
2021-04-09 05:37:51,053 autosynth [DEBUG] > Running: git log -1 --pretty=%at 68b8d39a3d5338ed5ff068ad8aa59fb271ef088b
2021-04-09 05:37:51,056 autosynth [DEBUG] > Running: git log -1 --pretty=%at edbd72054bf9f71505782a94d9973e26866f7038
2021-04-09 05:37:51,059 autosynth [DEBUG] > Running: git log -1 --pretty=%at f6132d5e7eddc9d8f08bc4a1cca9234baee38c94
2021-04-09 05:37:51,062 autosynth [DEBUG] > Running: git log -1 --pretty=%at a22531f8364582f51485e4e9db85872d93514ab1
2021-04-09 05:37:51,065 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-09 05:37:51,068 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-09 05:37:51,071 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-09 05:37:51,074 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-09 05:37:51,077 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-09 05:37:51,080 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-09 05:37:51,083 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-09 05:37:51,086 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0f3469e88a4be9af4d9b1c4b6cf6fd3f8ac15588
2021-04-09 05:37:51,089 autosynth [DEBUG] > Running: git log -1 --pretty=%at fbc7b766ef43c75293b26159ecaf66cb3fb23770
2021-04-09 05:37:51,092 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-09 05:37:51,097 autosynth [DEBUG] > Running: git checkout fbc7b766ef43c75293b26159ecaf66cb3fb23770
Note: checking out 'fbc7b766ef43c75293b26159ecaf66cb3fb23770'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at fbc7b76 feat: add excludes and patch callback arguments to owlbot_main (#1032)
2021-04-09 05:37:51,201 autosynth [DEBUG] > Running: git branch -f autosynth-28
2021-04-09 05:37:51,205 autosynth [DEBUG] > Running: git checkout autosynth-28
Switched to branch 'autosynth-28'
2021-04-09 05:37:51,209 autosynth [INFO] > Running synthtool
2021-04-09 05:37:51,210 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-09 05:37:51,210 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/28/sponge_log.log
2021-04-09 05:37:51,214 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-09 05:37:51,442 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-28
nothing to commit, working tree clean
2021-04-09 05:37:51,592 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-09 05:37:51,597 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-09 05:37:51,975 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-09 05:37:52,041 autosynth [ERROR] > Synthesis failed
2021-04-09 05:37:52,041 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-09 05:37:52,049 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-09 05:37:52,054 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/12125ade-676d-42af-ae49-c078573e5b52/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
[DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-10 05:37:57,509 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-10 05:37:57,513 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-10 05:37:57,517 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-10 05:37:57,521 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-10 05:37:57,525 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-10 05:37:57,528 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-10 05:37:57,532 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0f3469e88a4be9af4d9b1c4b6cf6fd3f8ac15588
2021-04-10 05:37:57,536 autosynth [DEBUG] > Running: git log -1 --pretty=%at fbc7b766ef43c75293b26159ecaf66cb3fb23770
2021-04-10 05:37:57,540 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6444141a0ab125ff5579e0b697b41ccc500cbda7
2021-04-10 05:37:57,544 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b0e1592dd7d70b485e157ea4b3eb1704ecbd015
2021-04-10 05:37:57,548 autosynth [DEBUG] > Running: git log -1 --pretty=%at 1f5e6bc8dc8e3661ee550905fc070e55e1b6cea1
2021-04-10 05:37:57,552 autosynth [DEBUG] > Running: git log -1 --pretty=%at d9ddac83a22a600dd33854c9d835a4fe52284207
2021-04-10 05:37:57,556 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0a071b3460344886297a304253bf924aa68ddb7e
2021-04-10 05:37:57,560 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-10 05:37:57,566 autosynth [DEBUG] > Running: git checkout 0a071b3460344886297a304253bf924aa68ddb7e
Note: checking out '0a071b3460344886297a304253bf924aa68ddb7e'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 0a071b3 chore: add constraints file check for python samples (#1012)
2021-04-10 05:37:57,673 autosynth [DEBUG] > Running: git branch -f autosynth-33
2021-04-10 05:37:57,677 autosynth [DEBUG] > Running: git checkout autosynth-33
Switched to branch 'autosynth-33'
2021-04-10 05:37:57,681 autosynth [INFO] > Running synthtool
2021-04-10 05:37:57,681 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-10 05:37:57,681 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/33/sponge_log.log
2021-04-10 05:37:57,686 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-10 05:37:57,926 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-33
nothing to commit, working tree clean
2021-04-10 05:37:58,084 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-10 05:37:58,089 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-10 05:37:58,480 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-10 05:37:58,549 autosynth [ERROR] > Synthesis failed
2021-04-10 05:37:58,549 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-10 05:37:58,557 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-10 05:37:58,563 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/aad321ad-c931-4b99-9785-541e8f5ce0bd/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
[DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-11 05:37:54,739 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-11 05:37:54,742 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-11 05:37:54,744 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-11 05:37:54,747 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-11 05:37:54,750 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-11 05:37:54,753 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-11 05:37:54,756 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0f3469e88a4be9af4d9b1c4b6cf6fd3f8ac15588
2021-04-11 05:37:54,758 autosynth [DEBUG] > Running: git log -1 --pretty=%at fbc7b766ef43c75293b26159ecaf66cb3fb23770
2021-04-11 05:37:54,761 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6444141a0ab125ff5579e0b697b41ccc500cbda7
2021-04-11 05:37:54,764 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b0e1592dd7d70b485e157ea4b3eb1704ecbd015
2021-04-11 05:37:54,767 autosynth [DEBUG] > Running: git log -1 --pretty=%at 1f5e6bc8dc8e3661ee550905fc070e55e1b6cea1
2021-04-11 05:37:54,770 autosynth [DEBUG] > Running: git log -1 --pretty=%at d9ddac83a22a600dd33854c9d835a4fe52284207
2021-04-11 05:37:54,772 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0a071b3460344886297a304253bf924aa68ddb7e
2021-04-11 05:37:54,776 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-11 05:37:54,781 autosynth [DEBUG] > Running: git checkout 0a071b3460344886297a304253bf924aa68ddb7e
Note: checking out '0a071b3460344886297a304253bf924aa68ddb7e'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 0a071b3 chore: add constraints file check for python samples (#1012)
2021-04-11 05:37:54,884 autosynth [DEBUG] > Running: git branch -f autosynth-33
2021-04-11 05:37:54,887 autosynth [DEBUG] > Running: git checkout autosynth-33
Switched to branch 'autosynth-33'
2021-04-11 05:37:54,891 autosynth [INFO] > Running synthtool
2021-04-11 05:37:54,892 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-11 05:37:54,892 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/33/sponge_log.log
2021-04-11 05:37:54,895 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-11 05:37:55,116 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-33
nothing to commit, working tree clean
2021-04-11 05:37:55,259 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-11 05:37:55,263 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-11 05:37:55,612 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-11 05:37:55,666 autosynth [ERROR] > Synthesis failed
2021-04-11 05:37:55,666 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-11 05:37:55,674 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-11 05:37:55,678 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/db63003a-622a-43a9-9439-c95c374e2d8f/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
[DEBUG] > Running: git log -1 --pretty=%at 6d76df2138f8f841e5a5b9ac427f81def520c15f
2021-04-12 05:37:49,413 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-12 05:37:49,416 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-12 05:37:49,418 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-12 05:37:49,421 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-12 05:37:49,424 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-12 05:37:49,427 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-12 05:37:49,429 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0f3469e88a4be9af4d9b1c4b6cf6fd3f8ac15588
2021-04-12 05:37:49,432 autosynth [DEBUG] > Running: git log -1 --pretty=%at fbc7b766ef43c75293b26159ecaf66cb3fb23770
2021-04-12 05:37:49,435 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6444141a0ab125ff5579e0b697b41ccc500cbda7
2021-04-12 05:37:49,438 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b0e1592dd7d70b485e157ea4b3eb1704ecbd015
2021-04-12 05:37:49,440 autosynth [DEBUG] > Running: git log -1 --pretty=%at 1f5e6bc8dc8e3661ee550905fc070e55e1b6cea1
2021-04-12 05:37:49,443 autosynth [DEBUG] > Running: git log -1 --pretty=%at d9ddac83a22a600dd33854c9d835a4fe52284207
2021-04-12 05:37:49,446 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0a071b3460344886297a304253bf924aa68ddb7e
2021-04-12 05:37:49,449 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-12 05:37:49,454 autosynth [DEBUG] > Running: git checkout 0a071b3460344886297a304253bf924aa68ddb7e
Note: checking out '0a071b3460344886297a304253bf924aa68ddb7e'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 0a071b3 chore: add constraints file check for python samples (#1012)
2021-04-12 05:37:49,557 autosynth [DEBUG] > Running: git branch -f autosynth-33
2021-04-12 05:37:49,561 autosynth [DEBUG] > Running: git checkout autosynth-33
Switched to branch 'autosynth-33'
2021-04-12 05:37:49,565 autosynth [INFO] > Running synthtool
2021-04-12 05:37:49,565 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-12 05:37:49,565 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/33/sponge_log.log
2021-04-12 05:37:49,568 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-12 05:37:49,788 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-33
nothing to commit, working tree clean
2021-04-12 05:37:49,931 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-12 05:37:49,936 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-12 05:37:50,291 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-12 05:37:50,348 autosynth [ERROR] > Synthesis failed
2021-04-12 05:37:50,348 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-12 05:37:50,356 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-12 05:37:50,361 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/3a660f13-c5e0-44f8-9ffc-ac9736ca0e61/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
f520c15f
2021-04-13 05:37:49,262 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-13 05:37:49,265 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-13 05:37:49,268 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-13 05:37:49,271 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-13 05:37:49,274 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-13 05:37:49,276 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-13 05:37:49,279 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0f3469e88a4be9af4d9b1c4b6cf6fd3f8ac15588
2021-04-13 05:37:49,282 autosynth [DEBUG] > Running: git log -1 --pretty=%at fbc7b766ef43c75293b26159ecaf66cb3fb23770
2021-04-13 05:37:49,285 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6444141a0ab125ff5579e0b697b41ccc500cbda7
2021-04-13 05:37:49,288 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b0e1592dd7d70b485e157ea4b3eb1704ecbd015
2021-04-13 05:37:49,291 autosynth [DEBUG] > Running: git log -1 --pretty=%at 1f5e6bc8dc8e3661ee550905fc070e55e1b6cea1
2021-04-13 05:37:49,294 autosynth [DEBUG] > Running: git log -1 --pretty=%at d9ddac83a22a600dd33854c9d835a4fe52284207
2021-04-13 05:37:49,297 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0a071b3460344886297a304253bf924aa68ddb7e
2021-04-13 05:37:49,300 autosynth [DEBUG] > Running: git log -1 --pretty=%at 082e1ca0863b13ada8594fe91845380765da5b70
2021-04-13 05:37:49,303 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-13 05:37:49,308 autosynth [DEBUG] > Running: git checkout 082e1ca0863b13ada8594fe91845380765da5b70
Note: checking out '082e1ca0863b13ada8594fe91845380765da5b70'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 082e1ca fix: typo (#1037)
2021-04-13 05:37:49,315 autosynth [DEBUG] > Running: git branch -f autosynth-34
2021-04-13 05:37:49,319 autosynth [DEBUG] > Running: git checkout autosynth-34
Switched to branch 'autosynth-34'
2021-04-13 05:37:49,323 autosynth [INFO] > Running synthtool
2021-04-13 05:37:49,323 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-13 05:37:49,323 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/34/sponge_log.log
2021-04-13 05:37:49,327 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-13 05:37:49,552 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-34
nothing to commit, working tree clean
2021-04-13 05:37:49,698 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-13 05:37:49,703 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-13 05:37:50,061 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-13 05:37:50,112 autosynth [ERROR] > Synthesis failed
2021-04-13 05:37:50,112 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-13 05:37:50,120 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-13 05:37:50,126 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/85b646ac-2ded-46b2-9848-56aa79ac781b/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

Autosynth is still having trouble generating python-access-context-manager. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the python-access-context-manager API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
f520c15f
2021-04-14 05:37:44,959 autosynth [DEBUG] > Running: git log -1 --pretty=%at ff39353f34a36e7643b86e97724e4027ab466dc6
2021-04-14 05:37:44,962 autosynth [DEBUG] > Running: git log -1 --pretty=%at bf04d88353da12e9326236164dcb3150dfbfcff8
2021-04-14 05:37:44,965 autosynth [DEBUG] > Running: git log -1 --pretty=%at 063de45298fbdd88916018ba566c7ecd254b39ae
2021-04-14 05:37:44,968 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b5bf6d519b2d658d9f2e483d9f6f3d0ba8ee6bc
2021-04-14 05:37:44,971 autosynth [DEBUG] > Running: git log -1 --pretty=%at f4c3a57aeddad8ad720e075a9e8342f1f7b9b766
2021-04-14 05:37:44,974 autosynth [DEBUG] > Running: git log -1 --pretty=%at 705743e66f5c0b24a95f7f30619c9d3ef747b317
2021-04-14 05:37:44,977 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0f3469e88a4be9af4d9b1c4b6cf6fd3f8ac15588
2021-04-14 05:37:44,980 autosynth [DEBUG] > Running: git log -1 --pretty=%at fbc7b766ef43c75293b26159ecaf66cb3fb23770
2021-04-14 05:37:44,983 autosynth [DEBUG] > Running: git log -1 --pretty=%at 6444141a0ab125ff5579e0b697b41ccc500cbda7
2021-04-14 05:37:44,986 autosynth [DEBUG] > Running: git log -1 --pretty=%at 5b0e1592dd7d70b485e157ea4b3eb1704ecbd015
2021-04-14 05:37:44,989 autosynth [DEBUG] > Running: git log -1 --pretty=%at 1f5e6bc8dc8e3661ee550905fc070e55e1b6cea1
2021-04-14 05:37:44,992 autosynth [DEBUG] > Running: git log -1 --pretty=%at d9ddac83a22a600dd33854c9d835a4fe52284207
2021-04-14 05:37:44,995 autosynth [DEBUG] > Running: git log -1 --pretty=%at 0a071b3460344886297a304253bf924aa68ddb7e
2021-04-14 05:37:44,998 autosynth [DEBUG] > Running: git log -1 --pretty=%at 082e1ca0863b13ada8594fe91845380765da5b70
2021-04-14 05:37:45,001 autosynth [DEBUG] > Running: git checkout 166c54cd73d2cfac6d45df2a676389f252fd73e3
Note: checking out '166c54cd73d2cfac6d45df2a676389f252fd73e3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-14 05:37:45,006 autosynth [DEBUG] > Running: git checkout 082e1ca0863b13ada8594fe91845380765da5b70
Note: checking out '082e1ca0863b13ada8594fe91845380765da5b70'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 082e1ca fix: typo (#1037)
2021-04-14 05:37:45,111 autosynth [DEBUG] > Running: git branch -f autosynth-34
2021-04-14 05:37:45,115 autosynth [DEBUG] > Running: git checkout autosynth-34
Switched to branch 'autosynth-34'
2021-04-14 05:37:45,120 autosynth [INFO] > Running synthtool
2021-04-14 05:37:45,120 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2021-04-14 05:37:45,120 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/python-access-context-manager/34/sponge_log.log
2021-04-14 05:37:45,124 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2021-04-14 05:37:45,358 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py.
On branch autosynth-34
nothing to commit, working tree clean
2021-04-14 05:37:45,512 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
2021-04-14 05:37:45,517 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.coveragerc
.flake8
.github/CONTRIBUTING.md
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/header-checker-lint.yml
.github/release-please.yml
.github/snippet-bot.yml
.gitignore
.kokoro/build.sh
.kokoro/continuous/common.cfg
.kokoro/continuous/continuous.cfg
.kokoro/docker/docs/Dockerfile
.kokoro/docker/docs/fetch_gpg_keys.sh
.kokoro/docs/common.cfg
.kokoro/docs/docs-presubmit.cfg
.kokoro/docs/docs.cfg
.kokoro/populate-secrets.sh
.kokoro/presubmit/common.cfg
.kokoro/presubmit/presubmit.cfg
.kokoro/publish-docs.sh
.kokoro/release.sh
.kokoro/release/common.cfg
.kokoro/release/release.cfg
.kokoro/samples/lint/common.cfg
.kokoro/samples/lint/continuous.cfg
.kokoro/samples/lint/periodic.cfg
.kokoro/samples/lint/presubmit.cfg
.kokoro/samples/python3.6/common.cfg
.kokoro/samples/python3.6/continuous.cfg
.kokoro/samples/python3.6/periodic-head.cfg
.kokoro/samples/python3.6/periodic.cfg
.kokoro/samples/python3.6/presubmit.cfg
.kokoro/samples/python3.7/common.cfg
.kokoro/samples/python3.7/continuous.cfg
.kokoro/samples/python3.7/periodic-head.cfg
.kokoro/samples/python3.7/periodic.cfg
.kokoro/samples/python3.7/presubmit.cfg
.kokoro/samples/python3.8/common.cfg
.kokoro/samples/python3.8/continuous.cfg
.kokoro/samples/python3.8/periodic-head.cfg
.kokoro/samples/python3.8/periodic.cfg
.kokoro/samples/python3.8/presubmit.cfg
.kokoro/test-samples-against-head.sh
.kokoro/test-samples-impl.sh
.kokoro/test-samples.sh
.kokoro/trampoline.sh
.kokoro/trampoline_v2.sh
.pre-commit-config.yaml
.trampolinerc
CODE_OF_CONDUCT.md
CONTRIBUTING.rst
LICENSE
MANIFEST.in
docs/_static/custom.css
docs/_templates/layout.html
docs/conf.py.j2
docs/multiprocessing.rst
noxfile.py.j2
renovate.json
Skipping: samples/AUTHORING_GUIDE.md
Skipping: samples/CONTRIBUTING.md
scripts/decrypt-secrets.sh
scripts/readme-gen/readme_gen.py.j2
scripts/readme-gen/templates/README.tmpl.rst
scripts/readme-gen/templates/auth.tmpl.rst
scripts/readme-gen/templates/auth_api_key.tmpl.rst
scripts/readme-gen/templates/install_deps.tmpl.rst
scripts/readme-gen/templates/install_portaudio.tmpl.rst
setup.cfg
testing/.gitignore
2021-04-14 05:37:45,887 synthtool [ERROR] > Failed executing nox -s generate_protos:

nox > Error while collecting sessions.
nox > Sessions not found: generate_protos

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/python-access-context-manager/synth.py", line 30, in <module>
    s.shell.run(["nox", "-s", "generate_protos"])
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['nox', '-s', 'generate_protos']' returned non-zero exit status 3.
2021-04-14 05:37:45,942 autosynth [ERROR] > Synthesis failed
2021-04-14 05:37:45,942 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 166c54c fix: add create_key to protos (#25)
2021-04-14 05:37:45,950 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-14 05:37:45,956 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/6235f0b2-5b6c-4c00-b337-9691e9773aac/targets/github%2Fsynthtool;config=default/tests;query=python-access-context-manager;failed=false).

--- a/.gitignore
+++ b/.gitignore
@@ -59,3 +59,6 @@ pylintrc.test
 
 # Exclude python-asset which is copied for testing
 python-asset/
+
+# Exclude api-common-protos which is only needed for generation
+api-common-protos

--- a/noxfile.py
+++ b/noxfile.py
@@ -174,30 +174,40 @@ def test(session, library):
         system(session)
 
 
-# @nox.session(python="3.8")
-# def generate_protos(session):
-#     """Generates the protos using protoc.
-#
-#     THIS SESSION CAN ONLY BE RUN LOCALLY
-#     Clone `api-common-protos` in the parent directory
-#
-#     Some notes on the `google` directory:
-#     1. The `_pb2.py` files are produced by protoc.
-#     2. The .proto files are non-functional but are left in the repository
-#        to make it easier to understand diffs.
-#     3. The `google` directory also has `__init__.py` files to create proper modules.
-#        If a new subdirectory is added, you will need to create more `__init__.py`
-#        files.
-#     """
-#     session.install("grpcio-tools")
-#     protos = [str(p) for p in (Path(".").glob("google/**/*.proto"))]
+@nox.session(python="3.8")
+def generate_protos(session):
+    """Generates the protos using protoc.
 
-#     session.run(
-#         "python",
-#         "-m",
-#         "grpc_tools.protoc",
-#         "--proto_path=../api-common-protos",
-#         "--proto_path=.",
-#         "--python_out=.",
-#         *protos,
-#     )
+    Some notes on the `google` directory:
+    1. The `_pb2.py` files are produced by protoc.
+    2. The .proto files are non-functional but are left in the repository
+       to make it easier to understand diffs.
+    3. The `google` directory also has `__init__.py` files to create proper modules.
+       If a new subdirectory is added, you will need to create more `__init__.py`
+       files.
+    """
+    session.install("grpcio-tools")
+    protos = [str(p) for p in (Path(".").glob("google/**/*.proto"))]
+
+    # Clone googleapis/api-common-protos
+    api_common_protos = "api-common-protos"
+    try:
+        session.run("git", "-C", api_common_protos, "pull", external=True)
+    except nox.command.CommandFailed:
+        session.run(
+            "git",
+            "clone",
+            "--single-branch",
+            f"https://github.com/googleapis/{api_common_protos}",
+            external=True,
+        )
+
+    session.run(
+        "python",
+        "-m",
+        "grpc_tools.protoc",
+        "--proto_path=api-common-protos",
+        "--proto_path=.",
+        "--python_out=.",
+        *protos,
+    )

